synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -586,20 +586,33 @@ async def _precache_image_url(
             og: The Open Graph dictionary. This is modified with image information.
         """
         # If there's no image or it is blank, there's nothing to do.
-        if "og:image" not in og or not og["og:image"]:
+        if "og:image" not in og:
+            return
+
+        # Remove the raw image URL, this will be replaced with an MXC URL, if successful.
+        image_url = og.pop("og:image")
+        if not image_url:
             return
 
         # The image URL from the HTML might be relative to the previewed page,
         # convert it to an URL which can be requested directly.
-        image_url = og["og:image"]
         url_parts = urlparse(image_url)
         if url_parts.scheme != "data":
             image_url = urljoin(media_info.uri, image_url)
 
         # FIXME: it might be cleaner to use the same flow as the main /preview_url
         # request itself and benefit from the same caching etc.  But for now we
         # just rely on the caching on the master request to speed things up.
-        image_info = await self._handle_url(image_url, user, allow_data_urls=True)
+        try:
+            image_info = await self._handle_url(image_url, user, allow_data_urls=True)
+        except Exception as e:
+            # Pre-caching the image failed, don't block the entire URL preview.
+            logger.warning(
+                "Pre-caching image failed during URL preview: %s errored with %s",
+                image_url,
+                e,
+            )
```

Comment:
I don't love catching all exceptions here; we don't care about e.g. 404s and such, but maybe we should log at error if something else happens?

Reply:
How would you propose narrowing?

Reply:
I have no idea, I'm hoping someone else will have ideas! 😄 I unfortunately think we've eaten the original exception in the `SImpleHttpClient` though...and we *do* want to catch many different exceptions here (non-200 results, DNS, etc.)